.github/workflows/post_release_actions.yml

```
@@ -79,6 +79,78 @@ jobs:
           echo "Pull Request Number - ${{ steps.cpr.outputs.pull-request-number }}"
           echo "Pull Request URL - ${{ steps.cpr.outputs.pull-request-url }}"
 
+  # Updates the VERSION file to ensure it's always greater than the released version.
+  # If the current VERSION (e.g., v2.5.0-dev) is <= the released tag (e.g., v2.5.0),
+  # it increments the minor version from the released tag (e.g., v2.6.0-dev).
+  pr-update-version:
+    runs-on: [ubuntu-latest]
+    needs: check-version
+    if: ${{ needs.check-version.outputs.version != '' }}
+    steps:
+      - uses: actions/checkout@v5
+        with:
+          fetch-depth: 0
+          ref: main
+      - name: Update VERSION file
+        env:
+          RELEASED_VERSION: ${{ needs.check-version.outputs.version }}
+        shell: bash
+        run: |
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+
+          current_version=$(head -n1 VERSION)
+          echo "Released version: $RELEASED_VERSION"
+          echo "Current VERSION file: $current_version"
+
+          # Validate current VERSION format: vMAJOR.MINOR.0-dev
+          if [[ ! "$current_version" =~ ^v[0-9]+\.[0-9]+\.0-dev$ ]]; then
+            echo "ERROR: Invalid VERSION format. Expected: vMAJOR.MINOR.0-dev"
+            exit 1
+          fi
+
+          # Validate released version format
+          if [[ ! "$RELEASED_VERSION" =~ ^v[0-9]+\.[0-9]+\.[0-9]+$ ]]; then
+            echo "ERROR: Invalid released version format. Expected: vMAJOR.MINOR.PATCH"
+            exit 1
+          fi
+
+          # Extract major.minor from released version
+          released_without_v=${RELEASED_VERSION#v}
+          IFS="." read -r released_major released_minor released_patch <<< "$released_without_v"
```

released_patch is not used.
```suggestion
          IFS="." read -r released_major released_minor _ <<< "$released_without_v"
```